db/per_partition_rate_limit_extension.hh

```diff
@@ -0,0 +1,42 @@
+/*
+ * Copyright (C) 2022-present ScyllaDB
+ */
+
+/*
+ * SPDX-License-Identifier: AGPL-3.0-or-later
+ */
+
+#pragma once
+
+#include "db/per_partition_rate_limit_options.hh"
+
+namespace db {
+
+class per_partition_rate_limit_extension : public schema_extension {
+    per_partition_rate_limit_options _options;
+public:
+    static constexpr auto NAME = "per_partition_rate_limit";
+
+    per_partition_rate_limit_extension() = default;
+    per_partition_rate_limit_extension(const per_partition_rate_limit_options& opts) : _options(opts) {}
+
+    explicit per_partition_rate_limit_extension(const std::map<sstring, sstring>& tags) : _options(tags) {}
+    explicit per_partition_rate_limit_extension(const bytes& b) : _options(deserialize(b)) {}
+    explicit per_partition_rate_limit_extension(const sstring& s) {
+        throw std::logic_error("Cannot create per partition rate limit info from string");
+    }
+
+    bytes serialize() const override {
+        return ser::serialize_to_buffer<bytes>(_options.to_map());
+    }
+    static std::map<sstring, sstring> deserialize(const bytes_view& buffer) {
+        return ser::deserialize_from_buffer(buffer, boost::type<std::map<sstring, sstring>>());
```

Comment:
Does this mean that this shows up in the schema as a blob?

I prefer a traditional serialization into a dedicated map column, so the schema can still be understood by people.

Reply:
Yes, it's serialized as a blob:

```
cqlsh> select extensions from system_schema.tables where table_name = 't' allow filtering;

 extensions
--------------------------------------------------------------------------------------------------------
 {'per_partition_rate_limit': 0x01000000140000006d61785f72656164735f7065725f7365636f6e6403000000313030}

(1 rows)
```

The type of the `extensions` column is `frozen<map<text, blob>>`. Even if we serialize it e.g. as a json, cqlsh will show ascii bytes. Should I change the serialization, or rather keep it as it is for consistency (all existing schema extensions use `ser::deserialize_from_buffer`)?

Reply:
The question is whether this should be a schema extension or not. Most times when we extend the schema we add columns.

Reply:
Schema extensions bypass the need of introducing new schema features at a (usually negligible) runtime cost, so I'm generally a fan, as long as parsing data from these features is not on the hot path.

Reply:
By the way, if we take https://github.com/scylladb/scylla/issues/9571 into consideration, it will become the server's responsibility to produce DESCRIBE results. Thanks to that, we'll be able to present all such blobs in human-readable form.